inference-engine/src/vpu/graph_transformer/src/stages/dynamic_shape_resolver.cpp

```diff
@@ -50,6 +50,11 @@ void FrontEnd::parseDSR(const Model& model, const ie::CNNLayerPtr& layer, const
         // Create the second output with shape in case of dynamic output
         const auto& shapeOutput = model->addOutputData(dataOutput->name() + "@shape", shape->desc());
 
+        bindData(shapeOutput, shape->origData());
+        for (const auto& shapeConsumersEdges : shape->consumerEdges()) {
+            model->replaceStageInput(shapeConsumersEdges, shapeOutput);
+        }
```

Comment:
Do we need this? To replace `shape` with `shapeOutput` we did `model->replaceStageOutput(shapeProducerEdge, shapeOutput);`. Is it enough?

Reply:
In theory, he should. But I look at the implementation of the method and see no replacing consumer edges to the new output.
So replaceStageOutput is not enough. Consumers of this shape (Broadcast for example) still have old input data which was removed as unused so nullptr is there.

Reply:
By the way, I see the same code (with iterating over consumers) in the other passes